hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/TransactionProcessor.java

```diff
@@ -160,6 +162,23 @@ private HederaEvmTransactionResult processTransactionWithParties(
         return safeCommit(result, transaction, updater, feesOnlyUpdater, context, config);
     }
 
+    private InvolvedParties safeComputeInvolvedParties(
+            @NonNull final HederaEvmTransaction transaction,
+            @NonNull final HederaWorldUpdater updater,
+            @NonNull final Configuration config,
+            @NonNull final HederaEvmContext context) {
+        try {
+            return computeInvolvedParties(transaction, updater, config);
+        } catch (HandleException e) {
+            // Aborted ContractCall transactions before processing in the evm result in charging all gas sent
+            if (transaction.isContractCall()) {
+                gasCharging.chargeGasForAbortedTransaction(transaction.senderId(), context, updater, transaction);
+                throw new AbortException(e.getStatus(), transaction.senderId());
```

Comment:
Up there in `ContextTransactionProcessor` it was necessary to immediately commit the `rootProxyWorldUpdater` after calling `chargeGasForAbortedTransaction`.

Is a commit of the `updater` necessary here? (If not, where is it done (for my information))

Reply:
[Here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java#L118).  This `catch` does not encompass the exception possibly thrown up above.